java-demo/src/main/java/org/pingcap/ticdc/cdc/TicdcEventFilter.java

```diff
@@ -0,0 +1,42 @@
+package org.pingcap.ticdc.cdc;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * 1.每个partition需要为每个表记录一个tbl-max-ts，小于这个值的为重复数据（基于row change event的TS）
+ * 2.每个partition同时还要记录resolved-max-ts，小于这个值的为重复数据（基于resolved event的TS）
+ * 3.以上两部分逻辑结合，理论上就能做到完全避免重复
+ *
+ * @NotThreadSafe
+ */
+public class TicdcEventFilter {
+    //    tablename_partition, rowChangeMaxTS
+    private Map<String, Long> tableMaxTSMap = new HashMap<>();
+    private Map<Integer, Long> resolveMaxTSMap = new HashMap<>();
+
+    public boolean filter(long rowChangeTS, int partition, String tableName) {
+        String mapKey = tableName + "_" + partition;
+        Long rowChangeMaxTS = tableMaxTSMap.get(mapKey);
+        if (rowChangeMaxTS == null) {
+            tableMaxTSMap.put(mapKey, rowChangeMaxTS);
+        } else {
+            if (rowChangeTS < rowChangeMaxTS) {
```

Comment:
```suggestion
            if (rowChangeTS <= rowChangeMaxTS) {
```
Is it also a duplicated event If the `rowChangeTS` equals to `rowChangeMaxTS`